test/e2e/storage/persistent_volumes.go

```diff
@@ -313,6 +313,32 @@ var _ = utils.SIGDescribe("PersistentVolumes", func() {
 				pvc = nil
 			})
 		})
+
+		ginkgo.Context("delete pv prior", func() {
+			ginkgo.BeforeEach(func() {
+				ginkgo.AfterEach(func() {
+					framework.Logf("AfterEach: Cleaning up test resources.")
+					if errs := e2epv.PVPVCCleanup(c, ns, pv, pvc); len(errs) > 0 {
+						framework.Failf("AfterEach: Failed to delete PVC and/or PV. Errors: %v", utilerrors.NewAggregate(errs))
+					}
+				})
+			})
+
+			// Waiting for the pvc and pvc bounded and then delete the pv prior
+			ginkgo.It("to exercise deletion of PV prior to deletion of PVC for a Bound PV-PVC pair", func() {
+				pv, pvc, err = e2epv.CreatePVPVC(c, f.Timeouts, pvConfig, pvcConfig, ns, false)
+				framework.ExpectNoError(err)
+				// 1. verify that the PV and PVC have bound correctly
+				ginkgo.By("Validating the PV-PVC binding")
+				framework.ExpectNoError(e2epv.WaitOnPVandPVC(c, f.Timeouts, ns, pv, pvc))
+
+				// 2. delete PV prior to deletion of PVC for a Bound PV-PVC pair
+				ginkgo.By("Wating for the pv deleted")
+				framework.ExpectNoError(e2epv.WaitForPersistentVolumeDeleted(c, pv.Name, 2*time.Second, 60*time.Second),
```

Comment:
@xing-yang only wait to the pv deleted,  it will hangs on "deleting the pv", after it,  the PVPVCClean() function will clean all resources.

Reply:
This function WaitForPersistentVolumeDeleted only waits for PV deletion. It does not deletes the PV itself.
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/pv/pv.go#L850-L864

We need a function that actually goes to delete PV from API server first, before trying to delete PV.